kyo-core/shared/src/main/scala/kyo/Retry.scala

```diff
@@ -41,18 +41,17 @@ object Retry:
             SafeClassTag[E],
             Frame
         ): A < (Async & Abort[E] & S) =
-            Loop(schedule) { schedule =>
```

Comment:
Since in most cases the operation succeeds and doesn't retry, using regular recursion saves a few `Loop`-related allocations in the happy path. It's also why the code doesn't get the `Clock` first to reuse on each iteration.